vpr/src/route/spatial_route_tree_lookup.cpp

```diff
@@ -78,7 +78,7 @@ size_t grid_to_bin_y(size_t grid_y, const SpatialRouteTreeLookup& spatial_lookup
 bool validate_route_tree_spatial_lookup(t_rt_node* rt_node, const SpatialRouteTreeLookup& spatial_lookup) {
     auto& device_ctx = g_vpr_ctx.device();
     const auto& rr_graph = device_ctx.rr_graph;
-    auto& rr_node = device_ctx.rr_nodes[rt_node->inode];
+    auto& rr_node = rr_graph.rr_nodes()[rt_node->inode];
```

Comment:
Since we have finished all the APIs, you can consider to replace this line by 

```
RRNodeId rr_node = rt_node->inode;
```
 And also adapt the rest codes in this function, ``rr_node.id()`` -> ``rr_node``